port/zephyr/CMakeLists.txt

```diff
@@ -5,6 +5,8 @@ endif()
 zephyr_include_directories(include)
 zephyr_include_directories(../../src/include)
 
+add_subdirectory_ifdef(CONFIG_GOLIOTH_SAMPLE_COMMON ../../examples/zephyr/common CMAKE_BINARY_DIR)
```

Comment:
This creates `modules/golioth-firmware-sdk/CMAKE_BINARY_DIR` in build directory. Was it intended?

Reply:
Second thing. Maybe let's replace `../../` (which is actually pointing at root directory, with `${ZEPHYR_GOLIOTH_FIRMWARE_SDK_MODULE_DIR}/` (to be verified if that is the symbol that can be used).

Reply:
> This creates modules/golioth-firmware-sdk/CMAKE_BINARY_DIR in build directory. Was it intended?

No, that's a mistake. I added this because of the following build error when I don't specify a binary_dir:

```
CMake Error at /home/mike/golioth-compile/golioth-firmware-sdk/zephyr/cmake/modules/extensions.cmake:1637 (add_subdirectory):                                                 
  add_subdirectory not given a binary directory but the given source                                                                                                          
  directory                                                                                                                                                                   
  "/home/mike/golioth-compile/golioth-firmware-sdk/modules/lib/golioth-firmware-sdk/examples/zephyr/common"                                                                   
  is not a subdirectory of                                                                                                                                                    
  "/home/mike/golioth-compile/golioth-firmware-sdk/modules/lib/golioth-firmware-sdk/port/zephyr".                                                                             
  When specifying an out-of-tree source a binary directory must be explicitly                                                                                                 
  specified.                                                                                                                                                                  
Call Stack (most recent call first):                                                                                                                                          
  /home/mike/golioth-compile/golioth-firmware-sdk/modules/lib/golioth-firmware-sdk/port/zephyr/CMakeLists.txt:8 (add_subdirectory_ifdef)   
```

Is it more appropriate to use the following?

```
add_subdirectory_ifdef(CONFIG_GOLIOTH_SAMPLE_COMMON
    ${ZEPHYR_GOLIOTH_FIRMWARE_SDK_MODULE_DIR}/examples/zephyr/common common)
```

Or is there a better Cmake command for conditionally including that directory in the build?

Reply:
```
add_subdirectory_ifdef(CONFIG_GOLIOTH_SAMPLE_COMMON
    ${ZEPHYR_GOLIOTH_FIRMWARE_SDK_MODULE_DIR}/examples/zephyr/common common)
```
seems fine. Or maybe `examples_zephyr_common` instead of `common` just to make sure there are no conflicts with other directories.

The reason you've seen an error is because you provided relative directory path. In that case CMake requires you to provide explicitly what is the build directory (instead of using build/binary directory name based on source directory name). This makes sense. Consider you have `add_subdirectory(../common ...)` and `add_subdirectory(../../common ...)`. In that case there would be a conflict and build objects from both directories would land in the same build/binary directory. So keep that in mind when providing explicitly what the build/binary directory should be.

Reply:
Great, thanks for the advice. I've changed this to:

```
add_subdirectory_ifdef(CONFIG_GOLIOTH_SAMPLE_COMMON
    ${ZEPHYR_GOLIOTH_FIRMWARE_SDK_MODULE_DIR}/examples/zephyr/common examples_zephyr_common)
```